src/test/ui/expr/malformed_closure/ruby_style_closure.stderr

```diff
@@ -0,0 +1,18 @@
+error[E0425]: cannot find value `x` in this scope
+  --> $DIR/ruby_style_closure.rs:5:14
+   |
+LL |         Some(x * 2)
+   |              ^ not found in this scope
+
+error[E0277]: expected a `FnOnce<({integer},)>` closure, found `Option<_>`
+  --> $DIR/ruby_style_closure.rs:2:22
+   |
+LL |     let p = Some(45).and_then({
+   |                      ^^^^^^^^ expected an `FnOnce<({integer},)>` closure, found `Option<_>`
```

Comment:
This should be pointing at the `Some(x * 2)`. (Not something to address in this PR)

Reply:
Done in https://github.com/rust-lang/rust/pull/88719